![New York City schoolbus](schoolbus.jpg)

Photo by [Jannis Lucas](https://unsplash.com/@jannis_lucas) on [Unsplash](https://unsplash.com).
<br>

Every year, American high school students take SATs, which are standardized tests intended to measure literacy, numeracy, and writing skills. There are three sections - reading, math, and writing, each with a **maximum score of 800 points**. These tests are extremely important for students and colleges, as they play a pivotal role in the admissions process.

Analyzing the performance of schools is important for a variety of stakeholders, including policy and education professionals, researchers, government, and even parents considering which school their children should attend. 

You have been provided with a dataset called `schools.csv`, which is previewed below.

You have been tasked with answering three key questions about New York City (NYC) public school SAT performance.

# Exploring NYC Public School Test Result Scores

This project analyzes SAT performance across New York City public schools using data on math, reading, and writing scores.

The analysis focuses on three questions:

1. Which NYC schools have the best math results?
2. Which schools have the highest combined SAT scores?
3. Which NYC borough has the largest variation in combined SAT scores?

## 1. Load and Explore the Data

First, import pandas and load the NYC public school SAT dataset.

In [31]:
import pandas as pd

schools = pd.read_csv("schools.csv")
schools.head()

,school_name,borough,building_code,average_math,average_reading,average_writing,percent_tested
0,"New Explorations into Science, Technology and ...",Manhattan,M022,657,601,601,NaN
1,Essex Street Academy,Manhattan,M445,395,411,387,78.9
2,Lower Manhattan Arts Academy,Manhattan,M445,418,428,415,65.1
3,High School for Dual Language and Asian Studies,Manhattan,M445,613,453,463,95.9
4,Henry Street School for International Studies,Manhattan,M056,410,406,381,59.7


In [32]:
schools.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 375 entries, 0 to 374
Data columns (total 7 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   school_name      375 non-null    object 
 1   borough          375 non-null    object 
 2   building_code    375 non-null    object 
 3   average_math     375 non-null    int64  
 4   average_reading  375 non-null    int64  
 5   average_writing  375 non-null    int64  
 6   percent_tested   355 non-null    float64
dtypes: float64(1), int64(3), object(3)
memory usage: 20.6+ KB


## 2. Schools with the Best Math Results

The maximum possible SAT Math score is 800.  
Schools with an average math score of at least 80% of the maximum score are considered to have the best math results.

Therefore, the minimum qualifying score is 640.

In [33]:
best_math_schools = (
    schools[schools["average_math"] >= 0.8 * 800]
    [["school_name", "average_math"]]
    .sort_values("average_math", ascending=False)
)


In [34]:
best_math_schools

,school_name,average_math
88,Stuyvesant High School,754
170,Bronx High School of Science,714
93,Staten Island Technical High School,711
365,Queens High School for the Sciences at York Co...,701
68,"High School for Mathematics, Science, and Engi...",683
280,Brooklyn Technical High School,682
333,Townsend Harris High School,680
174,High School of American Studies at Lehman College,669
0,"New Explorations into Science, Technology and ...",657
45,Eleanor Roosevelt High School,641


The results show the schools with an average math score of at least 640, ranked from highest to lowest.

## 3. Top 10 Schools Based on Combined SAT Scores

To compare overall SAT performance, I calculate a combined SAT score by adding the average math, reading, and writing scores for each school.

The schools are then sorted by their combined SAT score, and the top 10 schools are selected.

In [35]:
schools_new = schools.copy()

schools_new["total_SAT"] = (
    schools_new["average_math"]
    + schools_new["average_reading"]
    + schools_new["average_writing"]
)

In [36]:
schools_new.head()

,school_name,borough,building_code,average_math,average_reading,average_writing,percent_tested,total_SAT
0,"New Explorations into Science, Technology and ...",Manhattan,M022,657,601,601,NaN,1859
1,Essex Street Academy,Manhattan,M445,395,411,387,78.9,1193
2,Lower Manhattan Arts Academy,Manhattan,M445,418,428,415,65.1,1261
3,High School for Dual Language and Asian Studies,Manhattan,M445,613,453,463,95.9,1529
4,Henry Street School for International Studies,Manhattan,M056,410,406,381,59.7,1197


In [37]:
top_10_schools = (
    schools_new[["school_name", "total_SAT"]]
    .sort_values("total_SAT", ascending=False)
    .head(10)
)


In [38]:
top_10_schools

,school_name,total_SAT
88,Stuyvesant High School,2144
170,Bronx High School of Science,2041
93,Staten Island Technical High School,2041
174,High School of American Studies at Lehman College,2013
333,Townsend Harris High School,1981
365,Queens High School for the Sciences at York Co...,1947
5,Bard High School Early College,1914
280,Brooklyn Technical High School,1896
45,Eleanor Roosevelt High School,1889
68,"High School for Mathematics, Science, and Engi...",1889


These are the 10 schools with the highest combined average SAT scores across math, reading, and writing.

## 4. Borough with the Largest Variation in SAT Scores

Standard deviation measures how spread out the combined SAT scores are within each borough.

I will:
1. Group schools by borough.
2. Calculate the number of schools, average SAT score, and standard deviation of SAT scores.
3. Identify the borough with the largest standard deviation.
4. Round the numerical results to two decimal places.

In [39]:
schools_grpby_borough = (
    schools_new
    .groupby("borough")["total_SAT"]
    .agg(["count", "mean", "std"])
)

schools_grpby_borough

,count,mean,std
borough,,,
Bronx,98,1202.724490,150.393901
Brooklyn,109,1230.256881,154.868427
Manhattan,89,1340.134831,230.294140
Queens,69,1345.478261,195.252899
Staten Island,10,1439.000000,222.303596


In [40]:
largest_std_dev = (
    schools_grpby_borough[
        schools_grpby_borough["std"] == schools_grpby_borough["std"].max()
    ]
)

largest_std_dev

,count,mean,std
borough,,,
Manhattan,89,1340.134831,230.29414


In [41]:
largest_std_dev = (
    largest_std_dev
    .rename(
        columns={
            "borough": "borough",
            "count": "num_schools",
            "mean": "average_SAT",
            "std": "std_SAT"
        }
    )
    .round(2)
)

In [42]:
largest_std_dev

,num_schools,average_SAT,std_SAT
borough,,,
Manhattan,89,1340.13,230.29


### Finding

Manhattan has the largest standard deviation in combined SAT scores, with a standard deviation of 230.29 across 89 schools.

## Conclusion

The analysis identified the NYC schools with the strongest math performance and the top 10 schools based on combined SAT scores. Manhattan had the largest variation in combined SAT performance, with a standard deviation of 230.29 across 89 schools.